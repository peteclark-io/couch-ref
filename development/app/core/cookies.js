'use strict';

import Cookies from 'js-cookie';

export default const run = () => {
  var questions = Cookies.get('user.voted.questions');
  
};
