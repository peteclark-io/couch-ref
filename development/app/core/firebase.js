'use strict'

import firebase from 'firebase';
import {updateMatch, addMatch} from '../ducks/matches';
import {ready} from '../ducks/ready';

function createMatch(value){
  return {
    id: value.id,
    kickOff: value.kick_off,
    home: value.home,
    away: value.away,
    goalsHome: value.home_score,
    goalsAway: value.away_score,
    fullTime: value.full_time,
    referee: value.referee,
    homeLineup: value.home_lineup,
    awayLineup: value.away_lineup
  };
}

const data = (store) => {
  return {
    init: function(router, path){
      var config = {
        apiKey: "AIzaSyAoRaeRoKYx6Q_tuOVeK753OWmtuJEyQX8",
        authDomain: "couchref-9962e.firebaseapp.com",
        databaseURL: "https://couchref-9962e.firebaseio.com",
        storageBucket: "couchref-9962e.appspot.com",
      };

      firebase.initializeApp(config);

      return firebase.auth().signInAnonymously().catch(function(error) {
        var errorCode = error.code;
        var errorMessage = error.message;
        router.push('/error');
      }).then(() => {
        var database = firebase.database();
        var couchRef = database.ref('live-matches');

        couchRef.off();
        var dispatchAdd = function(data){
          store.dispatch(addMatch(createMatch(data.val())));
        };

        var dispatchUpdate = function(data){
          var value = data.val();
          store.dispatch(updateMatch(createMatch(data.val())));
        };

        couchRef.on('child_added', dispatchAdd);
        couchRef.on('child_changed', dispatchUpdate);

        store.dispatch(ready());
        router.push(path);
      });
    }
  };
}

export default data;
