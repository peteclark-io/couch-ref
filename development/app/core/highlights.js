'use strict';

import countries from './countries';
import _ from 'lodash';

const genders = ['Male', 'Female'];
const ageGroups = ['< 20', '21 - 30', '31 - 40', '41 - 50', '51 - 60', '60+'];
const populationRequirement = 10;

export function highlight(clubs, age, sex, location, club){
   var ageH = ageHighlights(age);
   var sexH = sexHighlights(sex);
   var countryH = countryHighlights(location);
   var clubH = clubHighlights(clubs, club);

   if (ageH.length === 0 && sexH.length === 0 && countryH.length === 0 && clubH.length === 0){
      return
   }

   var sorted = selectHighlight(clubH, ageH, sexH, countryH);

   var top = sorted[0];
   var bottom = sorted[sorted.length-1];

   createHeadline(top);
   createHeadline(bottom);

   if (top.headline > bottom.headline){
      return Object.assign(top, {
         blurb: createBlurb(top)
      });
   }

   return Object.assign(bottom, {
      blurb: createBlurb(bottom)
   });
};

const selectHighlight = (clubH, ageH, sexH, countryH) => {
   var union = _.sortBy(_.union(clubH, ageH, sexH, countryH), ['priority']).reverse();
   var priority = union[0].priority;
   var found = union.filter((p) => {return p.priority === priority;})
   return _.sortBy(found, ['percentage']);
};

const createHeadline = (highlight) => {
   var perc = highlight.percentage * 100;
   var yes = true;

   if (highlight.percentage < 0.1){
      perc = (1 - highlight.percentage) * 100;
      yes = false;
   }

   highlight.yes = yes;
   highlight.headline = perc;
}

const createBlurb = (highlight) => {
   switch (highlight.type) {
      case 'age':
         return 'of users aged ' + highlight.group + ' voted ' + (highlight.yes ? 'Yes' : 'No') + '!';
      case 'sex':
         return 'of ' + highlight.group + ' users voted ' + (highlight.yes ? 'Yes' : 'No') + '!';
      case 'location':
         return 'of users from ' + highlight.group + ' voted ' + (highlight.yes ? 'Yes' : 'No') + '!';
      case 'club':
         return 'of users who support ' + highlight.group + ' voted ' + (highlight.yes ? 'Yes' : 'No') + '!';
   }
   return '';
};

const clubHighlights = (clubs, clubD) => {
   var perc = clubs.map((club) => {
      if (!clubD[club]){
         return undefined;
      }

      var total = clubD[club].yes + clubD[club].no;
      return {
         priority: 4,
         type: 'club',
         group: club,
         percentage: clubD[club].yes / total
      };
   }).filter((r) => {
      if (r && (r.percentage > 0.9 || r.percentage < 0.1)){
         return r;
      }
   });

   return perc;
};

const countryHighlights = (location) => {
   var perc = countries.map((country) => {
      if (!location[country]){
         return undefined;
      }

      var total = location[country].yes + location[country].no;
      return {
         priority: 3,
         type: 'location',
         group: country,
         percentage: location[country].yes / total
      };
   }).filter((r) => {
      if (r && (r.percentage > 0.9 || r.percentage < 0.1)){
         return r;
      }
   });

   return perc;
};

const sexHighlights = (sex) => {
   var perc = genders.map((gender) => {
      if (!sex[gender]){
         return undefined;
      }

      var total = sex[gender].yes + sex[gender].no;
      return {
         priority: 1,
         type: 'sex',
         group: gender,
         percentage: sex[gender].yes / total
      };
   }).filter((r) => {
      if (r && (r.percentage > 0.9 || r.percentage < 0.1)){
         return r;
      }
   });

   return perc;
};

const ageHighlights = (age) => {
   var perc = ageGroups.map((group) => {
      if (!age[group]){
         return undefined;
      }

      var total = age[group].yes + age[group].no;
      return {
         priority: 2,
         type: 'age',
         group: group,
         percentage: age[group].yes / total
      };
   }).filter((r) => {
      if (r && (r.percentage > 0.9 || r.percentage < 0.1)){
         return r;
      }
   });

   return perc;
};
