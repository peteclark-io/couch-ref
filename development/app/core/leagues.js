'use strict';

const leagues = [
   {
      name: 'EFL Championship',
      shortName: 'EFL Championship'
   },
   {
      name: 'EFL League One',
      shortName: 'League One'
   },
   {
      name: 'EFL League Two',
      shortName: 'League Two'
   },
   {
      name: 'National League',
      shortName: 'National League'
   },
   {
      name: 'Other',
      shortName: 'Other'
   }
];

export default leagues;
