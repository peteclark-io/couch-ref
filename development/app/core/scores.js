'use strict';

export const overallScore = (score) => {
};

export const questionScore = (score) => {
   if(score > 0 && score < 1) {
      return 'Just about the right decision.';
   }

   if(score >= 1 && score < 3) {
      return 'Good decision!';
   }

   if(score >= 3 && score < 10) {
      return 'Excellent decision!';
   }

   if(score <= 0 && score > -1) {
      return 'Close, but not a good decision.';
   }

   if(score <= -1 && score > -2.5) {
      return 'Poor decision.';
   }

   if(score <= -2.5 && score > -4) {
      return 'Awful decision.';
   }

   if(score <= -4 && score > -6) {
      return 'Appalling decision.';
   }

   if(score <= -6) {
      return 'You\'ve had a nightmare.';
   }

   if(score <= -10) {
      return 'Should you even be a ref?';
   }
};

export const matchScore = (score) => {
   if(score > 0 && score < 1) {
      return 'Decent.';
   }

   if(score >= 1 && score < 3) {
      return 'Some Great Decisions!';
   }

   if(score >= 3 && score < 10) {
      return 'Superb!';
   }

   if(score <= 0 && score > -1) {
      return 'Not Bad.';
   }

   if(score <= -1 && score > -2.5) {
      return 'Room For Improvement.';
   }

   if(score <= -2.5 && score > -4) {
      return 'Poor.';
   }

   if(score <= -4 && score > -6) {
      return 'Terrible.';
   }

   if(score <= -6) {
      return 'A Nightmare.';
   }

   if(score <= -10) {
      return 'Should you even be a ref?';
   }
};
