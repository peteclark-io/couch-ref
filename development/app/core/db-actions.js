'use strict';

import firebase from 'firebase';

export const saveVote = (question, vote) => {
  var database = firebase.database();
  var stats = database.ref('/v0/live-statistics/' + question.id + '/simple');

  stats.transaction((simple) => {
    console.log(simple);
    vote ? simple.yes++ : simple.no++;
    return simple;
  });
};
