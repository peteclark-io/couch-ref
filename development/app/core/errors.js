'use strict';

import firebase from 'firebase';

export const error = (error) => {
   var database = firebase.database();
   var stats = database.ref('/v0/errors/' + question.id);
};
