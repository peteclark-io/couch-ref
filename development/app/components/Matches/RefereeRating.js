'use strict'

import React from 'react';
import {connect} from 'react-redux';

import _ from 'lodash';
import bootstrap from 'bootstrap/dist/css/bootstrap.css';
import classNames from 'classnames';

import {matchScore} from '../../core/scores';
import styles from './styles.css';

const RefereeRating = React.createClass({

   propTypes: {
      match: React.PropTypes.object,
      referee: React.PropTypes.object
   },

   render: function() {
      if (!this.props.match || !this.props.referee){
         return null;
      }

      var overall = this.props.referee.scores[this.props.match.id];
      console.log('Ref score for the match', overall)
      var title = matchScore(overall);

      return (
         <div className={styles['ref-rating']}>
            <h3>Referee Rating</h3>
            <h2>{title}</h2>
         </div>
      );
   }
});

const getLiveMatch = (state = {matches: []}, id) => {
   var filtered = _.filter(state.matches, {id: id});
   return filtered.length === 0 ? undefined : filtered[0];
};

const getReferee = (match, state = {referees: {}}) => {
   var ref = state.referees[match.referee];
   return ref ? ref : undefined;
};

const mapStateToProps = (state, ownProps) => {
   var match = getLiveMatch(state, ownProps.id);
   return {
      match: match,
      referee: getReferee(match, state)
   };
};

const LiveRefereeRating = connect(
   mapStateToProps
)(RefereeRating);

export default LiveRefereeRating;
