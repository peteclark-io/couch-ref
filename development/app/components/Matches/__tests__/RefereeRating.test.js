'use strict';

import React from 'react';
import {RefereeRating} from '../RefereeRating';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {matchScore} from '../../../core/scores';

it('Is null when it has no data.', () => {
   const rendered = mount(
      <RefereeRating />
   );
   expect(rendered.html()).toBeNull();

   rendered.setProps({match: {id: 'matchid'}})
   expect(rendered.html()).toBeNull();

   rendered.setProps({referee: {id: 'ref', scores: {}}})
   expect(rendered.html()).toBeNull();

   rendered.setProps({match: {id: 'matchid'}, referee: {id: 'ref', scores: {}}})
   expect(rendered.html()).toBeNull();
});

it('Shows the Ref rating when one is available', () => {
   const match = {id: 'matchid'}
   const referee = {scores: {matchid: 5}}

   const rendered = mount(
      <RefereeRating match={match} referee={referee} />
   );
   expect(rendered.html()).toMatchSnapshot();
});
