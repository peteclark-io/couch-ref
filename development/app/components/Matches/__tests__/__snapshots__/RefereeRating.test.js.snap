exports[`test Shows the Ref rating when one is available 1`] = `"<div class=\"ref-rating\"><h3>Referee Rating</h3><h2>Superb!</h2></div>"`;
